Mechanism management

anthony-walker opened the pull request:
This module was created to manage all mechanism files in the project. It is complete with CLI, script interface, and an accompanying test module. Also, it is currently set up to install with the included setup.py and pip. Documentation for this portion of the code is also up to date.

Commit:
Working on setup.py for CLI of mech management

Commit:
Worked out first version of mechanism management and tests

Commit:
Updated readme

Commit:
remove testing files

Commit:
Fix git ignore

kyleniemeyer commented on pyplume/__init__.py:
```diff
@@ -0,0 +1,2 @@
+from .mech import *
```
I don't think it's generally a good idea to use `import *`, because it clutters the namespace. Instead, maybe just use `from . import mech`, and then precede all functions with `mech.`

kyleniemeyer commented on pyplume/mech.py:
```diff
@@ -0,0 +1,88 @@
+        print("\t"+file)
+    print("done.\n")
+
+def mechCLI():
```
this function is missing a docstring

kyleniemeyer reviewed (changes requested):
I left some line comments; also, you probably don't need to explicitly include the mechanism files, since those come with cantera (which is a prereq)

Commit:
Fix init's

vickie-ngo commented on pyplume/mech.py:
```diff
@@ -0,0 +1,88 @@
+    print('done.\n')
+
+def mechFileList():
+    """Use this function to restore the original mechanism files supplied."""
```
More docstring details would make it more clear what each function does.

vickie-ngo reviewed (approved):
Looks great!

Commit:
Update doc strings

Commit:
Working on model generation - specifically creating combustor from cantera

Commit:
Sources

Commit:
Working on model generation. Currently working on connecting exhaust reactors. The idea is that boundary reactors will entrain air
and the next stages of reactors will receive some of that air based on mass follow between columns which could apparently mitigate
the need to transfer mass and up and down the columns. At least when using function to increase the number of reactors per column.

Commit:
README

Commit:
Trying new adjacency matrix idea

Commit:
Creating a template adj matrix took longer than I expected. Now that adjacency matrix is created. Connecting the reactors should be pretty simple

Commit:
Added linear expansion as class method, and started grid class method.

Commit:
Update readme
